src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -422,8 +422,12 @@ void refreshForeground()
 #endif
 			}
 
-			_foregroundChanged.Disposable =
-				Brush.AssignAndObserveBrush(Foreground, c => refreshForeground(), refreshForeground);
+			_foregroundChanged.Disposable = null;
+			if (Foreground?.SupportsAssignAndObserveBrush ?? false)
+			{
+				_foregroundChanged.Disposable =
+					Brush.AssignAndObserveBrush(Foreground, c => refreshForeground(), refreshForeground);
+			}
```

Comment:
Should we do something else if it's not supported? e.g, `ImageBrush.Subscribe()` as the original exception was suggesting?

Reply:
Not at this point. We'll tackle it with the other refactoring.